tree/ntuple/v7/test/ntuple_show.cxx

```diff
@@ -333,45 +333,58 @@ TEST(RNTupleShow, Objects)
 
 TEST(RNTupleShow, Collections)
 {
+   using ROOT::Experimental::RRecordField;
+   using ROOT::Experimental::RVectorField;
+
    std::string rootFileName{"test_ntuple_show_collection.root"};
    std::string ntupleName{"Collections"};
    FileRaii fileGuard(rootFileName);
    {
+      struct MyStruct {
+         int myInt;
+         float myFloat;
+      };
+
       auto model = RNTupleModel::Create();
-      auto collection_model = RNTupleModel::Create();
-      auto int_field = collection_model->MakeField<int>("myInt");
-      auto float_field = collection_model->MakeField<float>("myFloat");
-      auto collection = model->MakeCollection("collection", std::move(collection_model));
-      auto ntuple = RNTupleWriter::Recreate(std::move(model), ntupleName, rootFileName);
-      *int_field = 0;
-      *float_field = 10.0;
-      collection->Fill();
-      *int_field = 1;
-      *float_field = 20.0;
-      collection->Fill();
-      ntuple->Fill();
+      std::vector<std::unique_ptr<RFieldBase>> leafFields;
+      leafFields.emplace_back(std::make_unique<RField<int>>("myInt"));
+      leafFields.emplace_back(std::make_unique<RField<float>>("myFloat"));
+      auto recordField = std::make_unique<RRecordField>("_0", leafFields);
```

Comment:
Is there any requirement on the name (first argument)?  In particular does it 'have to' start with `_underscore`?

Reply:
That is the convention for the composed types in RNTuple. The sub fields (sub types) have the names `_0`, `_1`, ...

Reply:
User code will stray from that.  Does it matter?  If it does we need code to enforce the convention.  If it does not matter what spelling is used here, then we are good to go.

Reply:
The reading code will in fact accept any field name in this case (sub field of untyped collection). Probably better though to enforce the convention.